src/services/HttpService.ts

```diff
@@ -0,0 +1,160 @@
+import axios from './axios';
+
+/**
+ * Bare bones http service over an underlying implementation.
+ */
+
+/**
+ * Url replacements
+ */
+export type Replacements = Record<string, string>;
+
+/**
+ * Request types
+ */
+export type Request = {
+  url?: string; // override url to use
+  urlReplacements?: Replacements;
+  headers?: Record<string, any>;
+  params?: Record<string, string>; // query params
+};
+
+export type GetRequest = Request;
+
+export type PostRequest = Request & {
+  entity?: Record<string, any>;
+};
+
+export type PutRequest = PostRequest;
+
+export type PatchRequest = PostRequest;
+
+export type DeleteRequest = PostRequest;
+
+/**
+ * Response type
+ */
+export type Response = {
+  e?: Error;
+  error?: string;
+  data?: any;
+  statusCode?: number;
+  requestSucceeded: boolean;
+};
+
+/**
+ * Utility to populate an error string from server response
+ */
+const addError = (source: any, destination: any) => {
+  destination.error = source.error?.message;
+};
+
+/**
+ * Utility to process the response to an http request from the server
+ */
+const handleRequest = async (httpPromise: Promise<object>): Promise<Response> => {
+  const response: Response = {
+    requestSucceeded: false,
+  };
+
+  try {
+    const serverResponse: any = await httpPromise;
+    if (serverResponse.status) {
+      response.statusCode = serverResponse.status;
+    }
+    const data = serverResponse.data;
+    if (data?.status === 'error') {
+      addError(data, response);
+    } else {
+      response.requestSucceeded = true;
+      response.data = data;
+    }
+  } catch (e: any) {
+    response.e = e;
+    addError(e?.response?.data || {}, response);
+  }
+
+  return response;
+};
+
+/**
+ * Utility to replace placeholders with values in the url
+ */
+const replace = (url: string, request: Request) => {
+  const replacements = request.urlReplacements || {};
+  return Object.keys(replacements).reduce((accumulator, key) => {
+    return accumulator.replaceAll(key, replacements[key]);
+  }, request.url || url);
+};
+
+/**
+ * Service with bare bones http function calls
+ */
+function RequestsHandler(url: string = '') {
+  const get = async (request: GetRequest = {}): Promise<Response> => {
+    const { params, headers } = request;
+
+    return await handleRequest(axios.get(replace(url, request), { params, headers }));
+  };
+
+  const post = async (request: PostRequest = {}): Promise<Response> => {
+    const { entity, params, headers } = request;
+
+    return await handleRequest(axios.put(replace(url, request), entity, { params, headers }));
+  };
+
+  const put = async (request: PutRequest = {}): Promise<Response> => post(request);
```

Comment:
Curious about the choice to wrap `axios.put` into the `post` request type and then have `put` refer to `post`.. instead of wrapping it in the `put` and having `post` refer to that instead

Reply:
oops, good catch.. that entire function was put and i renamed it to post but missed the axios part.. will fix.

Reply:
fwiw - i have yet to use these functions in downstream services